src/main/java/sorald/processor/SerialVersionUidProcessor.java

```diff
@@ -0,0 +1,180 @@
+package sorald.processor;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import sorald.annotations.IncompleteProcessor;
+import sorald.annotations.ProcessorAnnotation;
+import spoon.reflect.declaration.CtClass;
+import spoon.reflect.declaration.CtElement;
+import spoon.reflect.declaration.CtField;
+import spoon.reflect.declaration.CtType;
+import spoon.reflect.declaration.ModifierKind;
+import spoon.reflect.reference.CtFieldReference;
+import spoon.reflect.reference.CtTypeReference;
+
+@IncompleteProcessor(
+        description =
+                "This processor ignores two corner cases: If the class has already a serialVersionUID with a non long type and if the class is not directly implementing serializable e.g. a upper class implements serializable.")
+@ProcessorAnnotation(
+        key = 2057,
+        description =
+                "Every class implementing Serializable should declare a static final serialVersionUID.")
+public class SerialVersionUidProcessor extends SoraldAbstractProcessor<CtClass<?>> {
+
+    private static final String DEFAULT_ID_VALUE = "1L";
+    private static final String SERIAL_VERSION_UID = "serialVersionUID";
+    private Set<ModifierKind> modifiers = Set.of(ModifierKind.STATIC, ModifierKind.FINAL);
+
+    @Override
+    protected boolean canRepairInternal(CtClass<?> candidate) {
+        Set<CtTypeReference<?>> superInterfaces = candidate.getSuperInterfaces();
+        if (isException(candidate)) {
+            return false;
+        }
+        CtTypeReference<?> serializable =
+                candidate.getFactory().createCtTypeReference(Serializable.class);
+        // check if the class implements serializable
+        if (superInterfaces.contains(serializable)) {
+
+            CtFieldReference<?> fieldReference = getSerialVersionUIDField(candidate);
+            // class implements serializable, lets check for serialVersionUID field
+            // the type of the field can be anything here, but should be long
+            if (fieldReference != null) {
+                CtField<?> field = fieldReference.getDeclaration();
+                // check the type
+                if (!isLongType(field)) {
+                    // serialVersionUID found but type is not long
+                    // => we cant fix it
+                    return false;
```

Comment:
> Even if it causes breakage, having a field like that is just plain wrong

It is pretty bad code for sure but sadly legal java. As we are in a somewhere strange corner case their could be some authors intend that the serialversionUID is always generated by the compiler and the field namespace is blocked for this reason.

I believe this case is never happens, but assuming we try to fix a class 
```java
public class Foo {
static final int serialVersionUID = 1;
}
```
We need to free the namespace for our long typed serialVersionUID, so we can either remove or rename the int type one.
#### Remove
Removing is just a no go if the variable is used, so we have to rename it. If the variable is unused the user just needs 2 sorald runs and the problem is fixed. First removes the variable second adds the UID.
#### Renaming
Renaming a public variable could produce a lot of changes in other classes the processor shouldn't touch. I think having a limited scope of a single class for each processor for each repair tasks simplifies a lot. If the variable is private we could rename it but then we try to fix the special case of a special case? I'm unsure if we even ever gonna find this case. Do you have this case somewhere in your experimentation data?

Reply:
> It is pretty bad code for sure but sadly legal java.

Legal, yes, but it's still wrong :). Kind of like this is legal Java, but will lead to a bad time: `int a = 1 / 0;`. With Sorald, we are very much concerned with fixing things that _are_ syntactically valid, yet are somehow incorrect. That's more or less the entire mission statement. So, I would say that removing the variable is a viable option in the scope of Sorald, although doing so unconditionally is not the ideal solution.

> I think having a limited scope of a single class for each processor for each repair tasks simplifies a lot.

It sure does, but for more complex repairs, we must go cross-file, and we have all intentions to do so in the near future. For example, I've my eyes set on repairing Sonar rules concerning class members that have too high visibility, which entails reducing their visibility and providing setters/getters if needed. That requires scanning the entire project for usages, and replacing direct member access with method calls.

An option for this rule we're discussing now would therefore be: remove the field if it's unused, rename it if it is used. As it's a comparatively simple refactoring from a conceptual standpoint, it would prove a good driver for prototyping cross-file repairs. We'd need a few helper methods for that that are outside the scope of implementing this processor (e.g. `rename(CtTypeMember)` and `findUsages(CtTypeMember)`), and so for now I suggest we simply keep the processor incomplete. Thoughts?

Reply:
I'm fine with keeping it local for now and solve the problem later.
I believe spoon has a refactoring for `rename(CtTypeMember)` and a filter for `findUsages(CtTypeMember)` already included, but I don't know about their quality state.

Reply:
Unfortunately, Spoon's high-level refactoring helpers only have renaming for methods and local variables at the moment. It might make sense to improve the refactoring utilities directly in Spoon: renaming a field is a pretty common operation.

I the code of this PR looks good now, I just want to have a look at if I can improve the test coverage to actually trigger the paths that I think we can remove once #387 is merged. I'll ping you once I've had the time to look it over.